src/services/services.ts

```diff
@@ -6259,6 +6379,193 @@ namespace ts {
 
                     return result[index];
                 }
+
+                function addReferenceToRelatedSymbol(node: Node, relatedSymbol: Symbol) {
+                    if (implementations) {
+                        const referenceEntries = getImplementationReferenceEntryForNode(node);

```

Comment:
consider passing the result array in and making getImplementationReferenceEntryForNode fill it directly instead of unpacking the array later on.